Merged app

ZixuanShu opened the pull request:
Sign up page and login page backend, filter and review page UI

ZixuanShu committed:
clickable filter

ZixuanShu committed:
Merge branch 'dev' of https://github.com/BU-Spark/se-scdao-mobile-flutter into Document_Screen

ZixuanShu committed:
simple review page UI done

ZixuanShu committed:
major update on filters functions, list and grid views

ZixuanShu committed:
Updated UI on review page

ZixuanShu committed:
search by name function added

ZixuanShu committed:
UI beautified

ZixuanShu committed:
merged app

billtu115 committed:
Update input validation

billtu115 committed:
Review list update

IanSaucy commented on lib/screens/login_screen.dart:
```diff
@@ -2,13 +2,45 @@ import 'package:flutter/gestures.dart';
+    Provider<LoginRepository>((ref) => LoginRepository());
+
+class LoginRepository {
+  var url = Uri.parse("http://192.168.0.44:8888/api/token");
```
Please set the base URL from an environment variable or similar method.

IanSaucy commented on lib/screens/login_screen.dart:
```diff
@@ -2,13 +2,45 @@ import 'package:flutter/gestures.dart';
+final loginRepositoryProvider =
+    Provider<LoginRepository>((ref) => LoginRepository());
+
+class LoginRepository {
```
Repository classes should be in their own file. We want to keep business logic(state manipulation,api calls etc) separate from the frontend code("presenter)

IanSaucy commented on lib/screens/login_screen.dart:
```diff
@@ -138,11 +172,20 @@ class _LoginScreenState extends State<LoginScreen> {
                   _validate = _text.text.isEmpty;
                   _emailvalidate = _emailtext.text.isEmpty;
                 });
+
+                res = await ref
+                    .read(loginRepositoryProvider)
+                    .login(_text.text, _emailtext.text);
+
+                if (res.statusCode == 200) {
+                  Navigator.of(context).pushNamed('DocumentPage');
+                  print("Yes!!");
+                }
```
You don't want this type of logic in your screen code. You should leverage something like `FutureProvider` to help with this and remove some of the boilerplate code. 

https://pub.dev/documentation/riverpod/latest/riverpod/FutureProvider-class.html

IanSaucy commented on lib/screens/signup_screen.dart:
```diff
@@ -52,39 +84,39 @@ class _SignupScreenState extends State<SignupScreen> {
-                ),
-              ),
-            ),
+            // Padding(
+            //   padding: const EdgeInsets.all(15.0),
+            //   child: TextField(
+            //     style: TextStyle(color: Colors.white),
+            //     controller: _emailtext,
+            //     decoration: InputDecoration(
+            //       errorText: _email
+            //           ? 'Username Can\'t Be Empty'
+            //           : 'Incorrect email format',
+            //       enabledBorder: UnderlineInputBorder(
+            //         borderSide: BorderSide(color: Colors.white),
+            //       ),
+            //       focusedBorder: UnderlineInputBorder(
+            //         borderSide: BorderSide(color: Colors.blueAccent),
+            //       ),
+            //       errorBorder: UnderlineInputBorder(
+            //         borderSide: BorderSide(color: Colors.redAccent),
+            //       ),
+            //       focusedErrorBorder: UnderlineInputBorder(
+            //         borderSide: BorderSide(color: Colors.redAccent),
+            //       ),
+            //       disabledBorder: UnderlineInputBorder(
```
Please remove if no longer needed

IanSaucy commented on lib/screens/signup_screen.dart:
```diff
@@ -209,6 +241,14 @@ class _SignupScreenState extends State<SignupScreen> {
                     _confirmP = true;
                 });
                 log('$_email');
+                res = await ref
+                    .read(signupRepositoryProvider)
+                    .signup(_userNametext.text, _passwordtext.text);
+
+                if (res.statusCode == 200) {
+                  Navigator.of(context).pushNamed('loginPage');
+                  print("Yes!!");
+                }
```
Same as above. Move logic out of this file and use something such as FutureProvider

IanSaucy commented on pubspec.yaml:
```diff
@@ -33,9 +34,9 @@ dependencies:
-  video_player: ^2.2.7
+  flutter_riverpod: ^1.0.0
+  http: ^0.13.4
+  provider: ^6.0.1
```
You should be using riverpod not provider! If you are not using please remove this dependency.

IanSaucy reviewed (changes requested):
Overall looks super great, awesome work both of you! 

Major thing is using more of the riverpod classes for helping manage state. 

The other request I have is to remove the use of setState and switch over to RiverPod 100%. This will allow you to remove business logic from the screen files to help keep things easier to read.

ZixuanShu committed:
basic UI for settings page

ZixuanShu committed:
Basic settingage down

ZixuanShu committed:
User initials avatar added

ZixuanShu committed:
Merge pull request #5 from BU-Spark/settings_Page

Settings page

toucham committed:
create services directory for api calling

- to follow OOP principles for less bugs

toucham committed:
static const route name for each screen

toucham committed:
refactor login screen according to figma

- add functionalities (api calling & navigation)
- create Theme & constants file later
- need to check for responsiveness & adaptiveness of screen page

toucham committed:
update dependencies to run ios simulator

toucham committed:
textcontroller + authscreen

- created custom widget for auth screens (sign up, login, forgot pwd and etc)
- added textcontroller and error handling
- loading animation

toucham committed:
add providers for state mngment

not tested yet

toucham committed:
Merge pull request #13 from toucham/refactor/login_screen

New login screen according to figma

toucham committed:
Add input validations & const for some widgets

toucham committed:
Connect FE to BE

toucham committed:
create caching logic for token auth

toucham committed:
Merge pull request #14 from toucham/refactor/auth_process

Refactor auth process

toucham committed:
init document screens

toucham committed:
finished views for document screen

toucham committed:
add filter logic & empty listview state

toucham committed:
routing

toucham committed:
camera screen

toucham committed:
Merge pull request #15 from toucham/feature/document_screen

Add document screen

toucham committed:
scan document screen

toucham committed:
display photos screen & update camera screen

toucham committed:
Merge pull request #16 from toucham/feature/display_screen

Add camera + display photos screens

toucham committed:
create popup menu for sorting

toucham committed:
Merge pull request #17 from toucham/feature/sort_files

create popup menu for sorting

toucham committed:
update README.md of SUM'22 works

toucham committed:
Update README.md

toucham committed:
Update README.md

toucham committed:
Update GIF for ux flows section in README.md

IanSaucy committed:
Updated ios setup stuff
